QoL update, fixed detached worker threads

mematthias opened the pull request:
- Fixed minimize_internal_with_timeout detached, joins worker thread
- Added TimeoutSignal
- Improved some vec.push operations

mematthias committed:
Fixed minimize_internal_with_timeout detached, joins worker thread
Added TimeoutSignal
Improved some vec.push operations

samildeli reviewed (changes requested):
In addition to the comments I left on the commit,
* Can you refactor to make timeout_signal parameters passed to the functions to be the last parameter?
* There is a time delay after the timeout before the thread is stopped. For example, when I give 100ms timeout, it takes 700ms for it to stop. Checking timeout_signal inside remove_dominating_terms and remove_dominated_implicants seems to mostly fix this. Can you address this?
* There seems to be precedence warnings in implicant.rs. Can you fix those while you're at it?

Thank you

mematthias committed:
Version 1.1.0 is now backwards compatible.

Code review comments:
- fixed timeout signal delay in remove_dominating_terms and remove_dominated_implicants
- timeout_signal parameter moved to the end
- fixed precedence warnings in implicant.rs
- fixed InvalidVariableCount error message format
- minimum Rust version is now 1.61, because of crate "thiserror"
- deprecated "unsafe" minimize operations replaced by new minimize operations

mematthias commented on src/lib.rs:
```diff
@@ -312,7 +315,7 @@ pub static DEFAULT_VARIABLES: [&str; 26] = [
 pub enum Error {
     /// The number of variables was less than 1 or greater than `DEFAULT_VARIABLES.len()`.
-    #[error("Invalid variable count: {0} (expected 1 <= variables.len() <= {})", DEFAULT_VARIABLES.len())]
+    #[error("Invalid variable count: {default_len} (expected 1 <= variables.len() <= {default_len})", default_len = DEFAULT_VARIABLES.len())]
```
Fixed

mematthias committed:
Removed use "use quine_mccluskey::MinimizeOptions"

mematthias committed:
Updated README.md

samildeli reviewed (changes requested):
Can you revert the changes to the API? I don't think it is necessary and we would need a major version increase. With the rest of the changes, I think we can get away with a patch version increase (1.0.1)

mematthias commented:
Before I revert the changes. Is a major release out of the question or do you simply not see it as necessary? Personally I have no preference, but a simple patch increase seems insufficient to me, as the changes could affect outside time-critical operations. I would say we should at least make a minor update

samildeli commented:
I see the API changes as unnecessary regardless of versioning.

For the minor update, I mean sure, but both minor and patch version updates are supposed to be non-breaking anyway, so we might as well make it a patch update. You could argue this should be a major version update especially because of the introduction of a rust-version field but I don't feel like doing that when there is no change to the public API. Also not many people are using the package currently, so I think it's okay.

mematthias committed:
- changed version from 1.1.0 to 1.0.1
- new and unneeded public API changes removed

mematthias committed:
applied rustfmt

mematthias commented:
So I undid the change, now the pull request should be ready. Many thanks for the crate :D
